includes/datavalues/SMW_DataValue.php

```diff
@@ -759,7 +759,7 @@ public function getHash() {
 	 *
 	 * @return boolean
 	 */
-	public function isNumeric() {
+	public function isNumeric(): bool {
```

Comment:
Return type narrowing to base classes or interfaces that are extended in other extensions require a major release. For this particular class, possibly Maps is the only affected extension, with its Geo DataValue.

Changes to ResultPrinter will have a wider impact

Reply:
We can always make a pragmatic exception to the rule, but would need to asses the impact of doing so. Or we could bump to 5.x.

Either way, probably good to first do https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/5366 before merging potentially disruptive changes.

Reply:
What do you suggest? Should we aim for a new major version in the short term, or is adding types like this too much effort? Otherwise I'll rather focus on missing types throwing deprecation notices in PHP 8.1 (e.g. #5370). But I suppose there is still a chance that by adding a missing type there we might still cause narrowing. I'm not familiar with all the downstream extensions, so that makes it a bit difficult to test.

Reply:
We should definitely throw out 7.3, maybe also 7.4.

New major version in the short term seems fine to me, but would still do a patch/minor release first.